mesonbuild/scripts/coverage.py

```diff
@@ -3,16 +3,48 @@
 
 from __future__ import annotations
 
-from mesonbuild import environment, mesonlib
+from mesonbuild import environment, mesonlib, coredata
 
 import argparse, re, sys, os, subprocess, pathlib, stat
 import typing as T
 
-def coverage(outputs: T.List[str], source_root: str, subproject_root: str, build_root: str, log_dir: str, use_llvm_cov: bool) -> int:
+def load_coredata(build_dir: str, options: 'argparse.Namespace') -> coredata.CoreData:
+    from mesonbuild.environment import Environment
+    from mesonbuild import mlog
+
+    scratch_dir = os.path.join(build_dir, Environment.private_dir)
+    log_dir = os.path.join(build_dir, Environment.log_dir)
+    info_dir = os.path.join(build_dir, Environment.info_dir)
+    os.makedirs(scratch_dir, exist_ok=True)
+    os.makedirs(log_dir, exist_ok=True)
+    os.makedirs(info_dir, exist_ok=True)
```

Comment:
This seems to be largely copied from environment.Environment; I would suggest refactoring that out for reuse instead, to avoid code duplication.

Reply:
This has now been taken care of by reordering the addition of the new command line options for the coverage script ahead of both the Ninja backend change and over this.